### DISCUSSION TITLE
Single Project Solution

### DISCUSSION THREAD

**Discussion by oscarjaergren** [unoplatform/uno#6539] [Q&A]
MAUI is doing a transition to a single file project solution and it's looking like a great way to improve Xamarin, is this possible to do the same for UNO?

**Comment by jeromelaban** [comment]
The set of supported platforms for Maui allow to place everything in a single project, something that Visual Studio (mac and Windows) does not support with the set of platforms supported by Uno. 

More specifically, WebAssembly and Skia (GTK/WPF) projects cannot be merged into a single project because the project system does not allow for project-specific discriminations, but only target-framework.

This being said, for .NET 6 we're merging all mobile targets into one, reducing the number of projects needed in a solution.

**Reply by oscarjaergren** [reply]
That's great to hear that all mobile targets will be merged. That will definintly help. Is it possible to alter the template at the same time to put all platforms into a "Platforms" folder like Maui does?

**Comment by robloo** [comment]
Duplicate of https://github.com/unoplatform/uno/discussions/6096

**Comment by HavenDV** [comment]
At the moment, you can try the single project solution within a separate repo - https://github.com/HavenDV/Uno.Sdk. This is a workload similar to tizen and this delivers`net7.0-webassembly;net7.0-gtk;net7.0-linux` to distinguish WebAssembly/Skia.Gtk/Skia.Linux.Framebuffer respectively. Skia.Wpf is implemented within `net7.0-windows` (this is an implicit indication for `net7.0-windows7`, which is different from the WinUI TargetFramework - `net7.0-windows10.x.x.x`).
Check the documentation for more details.
I would be glad to receive any feedback
